dispatches/case_studies/nuclear_case/nuclear_flowsheet.py

```diff
@@ -81,120 +109,157 @@ def build_ne_flowsheet(m, **kwargs):
-        destination=m.fs.translator.inlet
-    )
-
-    # Connect translator and mixer
-    m.fs.arc_translator_to_mixer = Arc(
-        source=m.fs.translator.outlet,
-        destination=m.fs.mixer.hydrogen_feed
-    )
-
-    # Connect mixer and h2 turbine
-    m.fs.arc_mixer_to_h2_turbine = Arc(
-        source=m.fs.mixer.outlet,
-        destination=m.fs.h2_turbine.compressor.inlet
-    )
+    m.fs.np_power_split.electricity[0].fix(np_capacity * 1e3)  # Converting MW to kW
+
+    if include_pem:
+        # Add PEM electrolyzer
+        m.fs.pem = PEM_Electrolyzer(property_package=m.fs.h2ideal_props)
+
+        # Connect the electrical splitter and PEM
+        m.fs.arc_np_to_pem = Arc(
+            source=m.fs.np_power_split.np_to_pem_port,
+            destination=m.fs.pem.electricity_in
+        )
+
+        if pem_capacity is not None:
+            m.fs.pem.electricity_in.electricity.setub(pem_capacity * 1e3)  # Converting MW to kW
+
+    else:
+        include_tank = False
+        include_turbine = False
+        m.fs.np_power_split.split_fraction["np_to_pem", 0].fix(0)
+
+    if include_tank:
+        # Add hydrogen tank
+        m.fs.h2_tank = SimpleHydrogenTank(property_package=m.fs.h2ideal_props)
+
+        # Connect the pem electrolyzer and h2 tank
+        m.fs.arc_pem_to_h2_tank = Arc(
+            source=m.fs.pem.outlet,
+            destination=m.fs.h2_tank.inlet
+        )
+
+        if tank_capacity is not None:
+            m.fs.h2_tank.tank_holdup_previous.setub(tank_capacity / mw_h2)
+
+    else:
+        include_turbine = False
+
+    if include_turbine:
+        # Add translator block
+        m.fs.translator = Translator(
+            inlet_property_package=m.fs.h2ideal_props,
+            outlet_property_package=m.fs.h2turbine_props,
+        )
+
+        # Add translator block constraints
+        m.fs.translator.eq_flow_hydrogen = Constraint(
+            expr=m.fs.translator.inlet.flow_mol[0] ==
+                m.fs.translator.outlet.flow_mol[0])
+
+        m.fs.translator.eq_temperature = Constraint(
+            expr=m.fs.translator.inlet.temperature[0] ==
+                m.fs.translator.outlet.temperature[0])
+
+        m.fs.translator.eq_pressure = Constraint(
+            expr=m.fs.translator.inlet.pressure[0] ==
+                m.fs.translator.outlet.pressure[0])
+
+        # Add mixer block
+        # using minimize pressure for all inlets and outlet of the mixer
+        # because pressure of inlets is already fixed in flowsheet,
+        # using equality will over-constrain
+        m.fs.mixer = Mixer(
+            momentum_mixing_type=MomentumMixingType.minimize,
+            property_package=m.fs.h2turbine_props,
+            inlet_list=["air_feed", "hydrogen_feed"],
+        )
+
+        # Add hydrogen turbine
+        m.fs.h2_turbine = HydrogenTurbine(
+            property_package=m.fs.h2turbine_props,
+            reaction_package=m.fs.reaction_params,
+        )
+
```

Comment:
I noticed when I was debugging the turbine model that the bounds for the flow_mol through the tank were pretty low. Depending on the size of your PEM, the flow might be getting close to the upper bounds. 

I ended up increasing all the bounds:

```
m.fs.h2_turbine.compressor.control_volume.properties_in[0.0].flow_mol.setub(1e7)
    m.fs.h2_turbine.compressor.control_volume.properties_out[0.0].flow_mol.setub(1e7)
    m.fs.h2_turbine.compressor.control_volume.properties_in[0.0].flow_mol_phase['Vap'].setub(1e7)
    m.fs.h2_turbine.compressor.control_volume.properties_out[0.0].flow_mol_phase['Vap'].setub(1e7)
    m.fs.h2_turbine.compressor.properties_isentropic[0.0].flow_mol_phase['Vap'].setub(1e7)
    m.fs.h2_turbine.compressor.properties_isentropic[0.0].flow_mol.setub(1e7)

    m.fs.h2_turbine.stoic_reactor.control_volume.properties_in[0.0].flow_mol_phase['Vap'].setub(1e7)
    m.fs.h2_turbine.stoic_reactor.control_volume.properties_in[0.0].flow_mol.setub(1e7)
    m.fs.h2_turbine.stoic_reactor.control_volume.properties_out[0.0].flow_mol.setub(1e7)
    m.fs.h2_turbine.stoic_reactor.control_volume.properties_out[0.0].flow_mol_phase['Vap'].setub(1e7)

    m.fs.h2_turbine.turbine.control_volume.properties_in[0.0].flow_mol_phase['Vap'].setub(1e7)
    m.fs.h2_turbine.turbine.control_volume.properties_in[0.0].flow_mol.setub(1e7)
    m.fs.h2_turbine.turbine.control_volume.properties_out[0.0].flow_mol.setub(1e7)
    m.fs.h2_turbine.turbine.control_volume.properties_out[0.0].flow_mol_phase['Vap'].setub(1e7)
    m.fs.h2_turbine.turbine.properties_isentropic[0.0].flow_mol_phase['Vap'].setub(1e7)
    m.fs.h2_turbine.turbine.properties_isentropic[0.0].flow_mol.setub(1e7)
```

Here's my latest flowsheet with updates: https://github.com/dguittet/dispatches/blob/5ed14ba53828ec8657a2fdcbdf67c0a9e628d97c/dispatches/case_studies/renewables_case/RE_flowsheet.py

Reply:
The default upper bound on all flows is 10,000. For the chosen conditions (H2_to_air ratio, deltaP, efficiencies), we can produce 75 MW - 80 MW when the flowrate is at its upper bound. If we want a bigger turbine, we can increase the upper bound to 1e5 (this should be sufficient to be produce 750 - 800 MW). I will update the upper bound in the code. 

FYI, if we want to update the bounds on all flow variables, we can update it at the property package level:
```python
from dispatches.properties.hturbine_ideal_vap import configuration as hturbine_config
hturbine_config["state_bounds"]["flow_mol"] = (0, 100, 1e5, pyunits.mol / pyunits.s)
m.fs.h2turbine_props = GenericParameterBlock(**hturbine_config)
```  
or directly update the dictionary in `dispatches.properties.hturbine_ideal_vap` file
This way, all the flow variables in the hydrogen turbine model will be defined with an upper bound of 1e5. Since you are also increasing the upper bound for the RE flowsheet, I will update the bound to 1e7 in the property package. This way, you don't have to update the bound manually.

Reply:
Cool. I didn't know how to do that at a broader level, so that's great.

Reply:
I updated the upper bound on flow_mol in the property package. Now, all flow variables in the turbine model will be defined with an upper bound of 1e7 by default.

Reply:
Increasing the upper bound makes one of the tests in the RE flowsheet fail. Could you please take a look at it?

Reply:
1e5 as the upper bound makes more sense than 1e7